pool.go

```diff
@@ -258,11 +271,20 @@ func (p *Pool) PrepareRetryableJob(job JobHandler) {
 	if dur <= 0 {
 		return
 	}
+	atomic.AddInt32(&p.numberOfRetryingJob, 1)
+	p.retryableWaitGroup.Add(1)
+
+	defer func() {
+		p.retryableWaitGroup.Done()
+		atomic.AddInt32(&p.numberOfRetryingJob, -1)
+	}()
+
 	timer := time.NewTimer(dur)
 	select {
 	case <-timer.C:
 		p.PrepareJobChan <- job
```

Comment:
Looking at this code, I can think of a case when `timer.C` happens at the same time of `p.ctx.Done`, pool can close this channel already while in `PrepareRetryableJob` the case `timer.C` is triggered before case `p.ctx.Done`. As a result, we get panic due to send to closed channel